From ada8ecc513c9821e8297c434ee3e8a68e1aa68f8 Mon Sep 17 00:00:00 2001
From: Daniel Krol <orblivion@gmail.com>
Date: Sat, 20 Nov 2021 12:55:36 -0500
Subject: [PATCH 1/6] Fix permission bug

Partial undo of e32ec365c78237a4c6ca740d8ffe8ee5421e3627. We don't want apool[0][0], we want numToAttribute[0][0]. Or so it seems and seems to work.
---
 index.js | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/index.js b/index.js
index 0cf7a81..9c3f63c 100644
--- a/index.js
+++ b/index.js
@@ -27,8 +27,8 @@ exports.padCopy = async (hookName, context) => {
 };
 
 exports.handleMessageSecurity = (hookName, context, callback) => {
-  const {message: {data: {apool} = {}} = {}} = context;
-  if (apool && apool[0] && apool[0][0] === 'comment') {
+  const {message: {data: {apool: {numToAttrib} = {}} = {}} = {}} = context;
+  if (numToAttrib && numToAttrib[0] && numToAttrib[0][0] === 'comment') {
     // Comment change, allow it to override readonly security model!!
     return callback(true);
   }
-- 
2.20.1

